backend/lucene/src/main/java/org/hibernate/search/backend/lucene/index/impl/LuceneIndexManagerBuilder.java

```diff
@@ -61,36 +60,38 @@ public IndexSchemaRootNodeBuilder getSchemaRootNodeBuilder() {
 	@Override
 	public LuceneIndexManagerImpl build() {
 		LuceneIndexModel model = null;
-		IndexWriter indexWriter = null;
+		IndexWriterHolder indexWriterHolder = null;
 		try {
 			model = schemaRootNodeBuilder.build( indexName );
-			indexWriter = createIndexWriter( model );
+			indexWriterHolder = createIndexWriterHolder( model );
 			return new LuceneIndexManagerImpl(
-					indexingBackendContext, searchBackendContext, indexName, model, indexWriter
+					indexingBackendContext, searchBackendContext, indexName, model, indexWriterHolder
 			);
 		}
 		catch (RuntimeException e) {
 			new SuppressingCloser( e )
 					.push( model )
-					.push( indexWriter );
+					.push( IndexWriterHolder::closeIndexWriter, indexWriterHolder );
 			throw e;
 		}
 	}
 
-	private IndexWriter createIndexWriter(LuceneIndexModel model) {
-		IndexWriterConfig indexWriterConfig = new IndexWriterConfig( model.getScopedAnalyzer() );
+	private IndexWriterHolder createIndexWriterHolder(LuceneIndexModel model) {
+		Directory directory;
 		try {
-			Directory directory = indexingBackendContext.createDirectory( indexName );
-			try {
-				return new IndexWriter( directory, indexWriterConfig );
-			}
-			catch (RuntimeException e) {
-				new SuppressingCloser( e ).push( directory );
-				throw e;
-			}
+			directory = indexingBackendContext.createDirectory( indexName );
 		}
 		catch (IOException | RuntimeException e) {
-			throw log.unableToCreateIndexWriter( getEventContext(), e );
+			throw log.unableToCreateIndexDirectory( getEventContext(), e );
+		}
+		try {
+			return indexingBackendContext.createIndexWriterHolder(
+					model.getIndexName(), directory, model.getScopedAnalyzer()
+			);
```

Comment:
Maybe we can group the calls `createDirectory` and `createIndexWriterHolder`, since they are performed on the same instance and directory is never used outside of the method.

Reply:
Right, done